File: mobile-client/android/Data/Message/Thread/MessageThreadGateway.kt
package app.mobiling.client.data.message.thread

import app.mobiling.client.contract.message.thread.MessageItemPayload
import app.mobiling.client.contract.message.thread.MessageThreadSummary
import app.mobiling.client.contract.message.thread.SendMessageRequest

/**
 * Marketing America Corp. Oleksandr Tishchenko
 */
interface MessageThreadGateway {
    suspend fun listThreads(): List<MessageThreadSummary>
    suspend fun listItems(threadId: String): List<MessageItemPayload>
    suspend fun sendMessage(request: SendMessageRequest): MessageItemPayload
}

File: mobile-client/ios/Data/Message/Thread/MessageThreadGateway.swift
import Foundation

// Marketing America Corp. Oleksandr Tishchenko
public protocol MessageThreadGateway: Sendable {
    func listThreads() async throws -> [MessageThreadSummary]
    func listItems(threadId: String) async throws -> [MessageItemPayload]
    func sendMessage(request: SendMessageRequest) async throws -> MessageItemPayload
}
